### PR TITLE
test/librbd: allow parallel runs of run-rbd-unit-tests

### PR DESCRIPTION
**Pull request by wjwithagen**
Running all tests sequential makes it the longest test of
`make check`, with each partial test taking around 500 sec.
Running 6 tests thus takes almost an hour.

Cut this down if ctest runs tests in parallel

Default behaviour of src/test/run-rbd-unit-tests.sh is kept:
  Without parameters the tests are run in sequence

To run unitttest_librbd with RBD_FEATURES, use `N` as parameter

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by dillaman** [comment]
500 seconds per test run!? On my machine it's only 22 seconds per test run. I really don't want to lose the ability for me to kick off all the RBD unit tests w/ a single command line. It would be very painful to have to run `ctest -V -R run-rbd-unit-tests-XYZ.sh` for each possible combination. Instead, perhaps keep the original functionality of the script if no argument was passed but have it skip running the tests if some new environment variable was set (via `run-make-check.sh`)?

**Comment by wjwithagen** [comment]
> perhaps keep the original functionality of the script if no argument was passed but have it skip running the tests if some new environment variable was set (via `run-make-check.sh`)?

@dillaman
I think that that is what I tried doing by seriously mangeling the script and cmake.
But instead of running `ctest -V -R run-rbd-unit-tests-XYZ.sh` you would run:
    `ctest -V -R run-rbd-unit-tests` 
and all matching tests would be run, even in parallel if your prefer.

**Comment by dillaman** [comment]
>  I think that that is what I tried doing by seriously mangeling the script and cmake.

I wouldn't want them to run in parallel in my environment -- it will make my life miserable trying to figure out which feature set caused the test to fail (unless you could somehow prefix each line output w/ the current feature set).

**Comment by wjwithagen** [comment]
> 
> 
> > I think that that is what I tried doing by seriously mangeling the script and cmake.
> 
> I wouldn't want them to run in parallel in my environment -- it will make my life miserable trying to figure out which feature set caused the test to fail (unless you could somehow prefix each line output w/ the current feature set).

@dillaman 
As long as you do not use `ctest -V -R run-rbd-unit-tests ` all tests are exected sequentially as they used to be.

Reason for splitting them out was exactly your argument not to run the different feature sets parallel from with the script. Doing it that way would indeed cause the problem you indicated.
Now you will get a nice report at the bottom which of the feature sets failed, and above is the verbose output.

Adding `--output-on-failure` would even only generate output for the failed tests.

File: src/test/run-rbd-unit-tests.sh [hunk 1]
```diff
@@ -1,17 +1,25 @@
 #!/usr/bin/env bash
-set -ex
+set -xe
```
Comment (dillaman) [inline src/test/run-rbd-unit-tests.sh]: Nit: why this change?

File: src/test/run-rbd-unit-tests.sh [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
+  unset RBD_FEATURES; unittest_librbd
+else 
+  TESTS="$*"
+fi
 
-for i in 0 1 61 109 127
+for i in ${TESTS}
 do
-    RBD_FEATURES=$i unittest_librbd
+        RBD_FEATURES=$i unittest_librbd
```
Comment (dillaman) [inline src/test/run-rbd-unit-tests.sh]: Nit: why the extra indentation?

File: src/test/run-rbd-unit-tests.sh (hunk 1 above)
Comment (wjwithagen) [reply src/test/run-rbd-unit-tests.sh]: @dillaman
Both not intentional, will revert 'm

**Comment by dillaman** [comment]
jenkins test docs
